app/services/books_service.rb

```diff
@@ -0,0 +1,13 @@
+class BooksService
+  include HTTParty
+  base_uri Rails.application.credentials.base_uri
+
+  def find_book(id)
+    path = '/api/books'
+    options = { query: { bibkeys: id, format: 'json', jscmd: 'data' } }
+    book = self.class.get(path, options)
+
+    keys = %i[title subtitle number_of_pages authors]
+    JSON.parse(book.body)[id].deep_symbolize_keys.slice(*keys)
```

Comment:
move this logic to another method like at `process_response` the function of find_book method is only find not process a data